### DISCUSSION TITLE
Running docker @QuarkusIntegrationTest inside of docker

### DISCUSSION THREAD

**Discussion by ennishol** [quarkusio/quarkus#37043] [Community]
Hello, 
I'm using a docker container to run all tests and push the image with jib on Jenkins. I also would like to test the final image with `@QuarkusIntegrationTest` but it does not work as the docker image I'm using doesn't have docker installed. 

How do you run docker `@QuarkusIntegrationTest`? Do you install the required java version on CI? Or do you install docker in the docker image running the tests? Any alternatives?
Thanks!

**Comment by bulldog98** [comment]
You can mount the docker binary and the docker socket into the container, if your host is a Linux and runs on the x84.

_Edit_: this is not secure

**Reply by ennishol** [reply]
@bulldog98  do you have any working example? I've tried  `docker  run -v /usr/bin/docker:/user/bin/docker:z -v /var/run/docker.sock:/var/run/docker.sock -u root -it --rm --name jdk-21 myimage` but it did not work. 
thanks!

**Reply by bulldog98** [reply]
For me `docker  run -v /usr/bin/docker:/usr/bin/docker:z -v /var/run/docker.sock:/var/run/docker.sock -u root -it --rm --name jdk-21 --entrypoint=/bin/bash eclipse-temurin:21-jdk` works, note I'm using a debian based distro and the image does that too. But I think the main problem is the missspelling of `/usr` in the first bind

**Reply by ennishol** [reply]
@bulldog98 Oh, thanks for spotting the typo!
On WSL also  `/usr/bin/com.docker.cli` is needed btw

**Reply by ennishol** [reply]
@bulldog98  still doesn't work with an integration test. Asked #37179
